pkg/apis/resourcemanager/v1alpha1/vendor_types.go

```diff
@@ -0,0 +1,201 @@
+// +kubebuilder:object:generate=true
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VendorProfileType defines the type of vendor profile
+// +kubebuilder:validation:Enum=person;business
+type VendorProfileType string
+
+const (
+	VendorProfileTypePerson   VendorProfileType = "person"
+	VendorProfileTypeBusiness VendorProfileType = "business"
+)
+
+// VendorStatusValue defines the current status of a vendor
+// +kubebuilder:validation:Enum=pending;active;rejected;archived
+type VendorStatusValue string
+
+const (
+	VendorStatusPending  VendorStatusValue = "pending"
+	VendorStatusActive   VendorStatusValue = "active"
+	VendorStatusRejected VendorStatusValue = "rejected"
+	VendorStatusArchived VendorStatusValue = "archived"
+)
+
+// CorporationType defines the type of corporation
+// This should reference a valid corporation type from CorporationTypeConfig
+// +kubebuilder:validation:Pattern=^[a-z0-9-]+$
+type CorporationType string
+
+// TaxIdType defines the type of tax identification
+// +kubebuilder:validation:Enum=SSN;EIN;ITIN;UNSPECIFIED
+type TaxIdType string
+
+const (
+	TaxIdTypeSSN         TaxIdType = "SSN"
+	TaxIdTypeEIN         TaxIdType = "EIN"
+	TaxIdTypeITIN        TaxIdType = "ITIN"
+	TaxIdTypeUnspecified TaxIdType = "UNSPECIFIED"
+)
+
+// Address represents a physical address
+type Address struct {
+	// Street address line 1
+	// +kubebuilder:validation:Required
+	Street string `json:"street"`
+
+	// Street address line 2 (optional)
+	// +optional
+	Street2 string `json:"street2,omitempty"`
+
+	// City
+	// +kubebuilder:validation:Required
+	City string `json:"city"`
+
+	// State or province
+	// +kubebuilder:validation:Required
+	State string `json:"state"`
+
+	// Postal or ZIP code
+	// +kubebuilder:validation:Required
+	PostalCode string `json:"postalCode"`
+
+	// Country
+	// +kubebuilder:validation:Required
+	Country string `json:"country"`
+}
+
+// TaxInfo represents tax-related information
+type TaxInfo struct {
+	// Type of tax identification
+	// +kubebuilder:validation:Required
+	TaxIdType TaxIdType `json:"taxIdType"`
+
+	// Tax identification number
+	// +kubebuilder:validation:Required
+	TaxId string `json:"taxId"`
+
+	// Country for tax purposes
+	// +kubebuilder:validation:Required
+	Country string `json:"country"`
+
+	// Tax document reference (e.g., W-9, W-8BEN)
+	// +kubebuilder:validation:Required
+	TaxDocument string `json:"taxDocument"`
+
+	// Whether tax information has been verified
+	// +kubebuilder:default=false
+	TaxVerified bool `json:"taxVerified"`
+
+	// Timestamp of tax verification
+	// +optional
+	VerificationTimestamp *metav1.Time `json:"verificationTimestamp,omitempty"`
```

Comment:
I'd consider a `VendorVerification` type that has a reference to the vendor and any information around that verification (who verified it, etc). This helps keep the vendor object scoped to just a vendor's information and other types are used to do verifications or add additional information. RBAC becomes much simpler by keeping these resources separate.